assets/js/search.js

```diff
@@ -0,0 +1,124 @@
+(function() {
+    'use stict';
+    var activityCards, find, mapCardToClassName, mapCardToHeading, hideCard,
+        showCard, filterCards, onSeachChange, onClearSearch, createSearchInput,
+        activityContainer, searchInput;
+
+    activityCards = [].slice.call(document.querySelectorAll('.activity-card'));
+
+    find = function(needle, haystack) {
+        var regex;
+
+        regex = new RegExp(needle);
+        return regex.test(haystack);
+    };
+
+    mapCardToClassName = function(card) {
+        return card.className;
+    };
+
+    mapCardToHeading = function(card) {
+        var heading;
+
+        heading = card.querySelector('h2').textContent;
+        return heading.toLowerCase();
+    };
+
+    hideCard = function(card) {
+        if (card.className.split(' ').indexOf('fade') === -1) {
+            card.className += ' fade';
+        }
+        // Bootstrap's .fade animates for 150 ms
+        setTimeout(function() { card.style.display = 'none'; }, 155);
+    };
+
+    showCard = function(card) {
+        // kudos https://gist.github.com/k-gun/c2ea7c49edf7b757fe9561ba37cb19ca
+        // resp. https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#Polyfill
+        if (card.className.split(' ').indexOf('fade') > -1) {
+            card.className = card.className.replace(new RegExp('(^| )fade( |$)'), '');
+        }
+        // In order to kick in after async hideCard calls
+        // 300 ms is a magic limit, where effect becomes notable to humans
+        setTimeout(function() { card.style.display = ''; }, 300);
+    };
+
+    filterCards = function(searchTerm) {
+        var classNames, headings;
+
+        searchTerm = searchTerm.toLowerCase();
+        classNames = activityCards.map(mapCardToClassName);
+        classNames = classNames.filter(function(className) {
+            return find(searchTerm, className);
+        });
+
+        headings = activityCards.map(mapCardToHeading);
+        headings = headings.filter(function(heading) {
+            return find(searchTerm, heading);
+        });
+
+        activityCards.forEach(function(card) {
+            if (find(searchTerm, mapCardToClassName(card))) {
```

Comment:
just an idea: to make this if/else easier, can we rewrite the find method so it takes an array?